platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ConfigurationContainerExtensions.kt

```diff
@@ -84,3 +86,114 @@ private constructor(
+ *
+ * @return A provider which creates a new resolvable configuration.
+ *
+ * @throws InvalidUserDataException If a configuration with the given name already exists in this container.
+ *
+ * @since 8.5
+ */
+@Incubating
+fun ConfigurationContainer.resolvable(action: Action<in ResolvableConfiguration>): ConfigurationProvider<ResolvableConfiguration> =
+    ConfigurationProvider { name -> resolvable(name, action) }
+
+
+/**
+ * Registers a [ConsumableConfiguration] via [consumable].
+ *
+ * @return A provider which creates a new consumable configuration.
+ *
+ * @throws InvalidUserDataException If a configuration with the given name already exists in this container.
+ *
+ * @since 8.5
+ */
+@get:Incubating
+val ConfigurationContainer.consumable: ConfigurationProvider<ConsumableConfiguration>
+    get() = ConfigurationProvider { name -> consumable(name) }
+
+
+/**
+ * Registers a [ConsumableConfiguration] via [consumable] and then
+ * configures it with the provided action.
+ *
+ * @param action The action to apply to the configuration.
+ *
+ * @return A provider which creates a new consumable configuration.
+ *
+ * @throws InvalidUserDataException If a configuration with the given name already exists in this container.
+ *
+ * @since 8.5
+ */
+@Incubating
+fun ConfigurationContainer.consumable(action: Action<in ConsumableConfiguration>): ConfigurationProvider<ConsumableConfiguration> =
+    ConfigurationProvider { name -> consumable(name, action) }
+
+
+/**
+ * Registers a [DependencyScopeConfiguration] via [dependencyScope].
+ *
+ * @return A provider which creates a new dependency scope configuration.
+ *
+ * @throws InvalidUserDataException If a configuration with the given name already exists in this container.
+ *
+ * @since 8.5
+ */
+@get:Incubating
+val ConfigurationContainer.dependencyScope: ConfigurationProvider<DependencyScopeConfiguration>
+    get() = ConfigurationProvider { name -> dependencyScope(name) }
+
+
+/**
+ * Registers a [DependencyScopeConfiguration] via [dependencyScope] and then
+ * configures it with the provided action.
+ *
+ * @param action The action to apply to the configuration.
+ *
+ * @return A provider which creates a new dependency scope configuration.
+ *
+ * @throws InvalidUserDataException If a configuration with the given name already exists in this container.
+ *
+ * @since 8.5
+ */
+@Incubating
+fun ConfigurationContainer.dependencyScope(action: Action<in DependencyScopeConfiguration>): ConfigurationProvider<DependencyScopeConfiguration> =
+    ConfigurationProvider { name -> dependencyScope(name, action) }
+
+
+/**
+ * Provides access to the [NamedDomainObjectProvider] for the element of the given
+ * property name from the configuration container via a delegated property.
+ *
+ * @since 8.5
+ */
+@Incubating
+class ConfigurationProvider<C : Configuration> internal constructor(private val createProvider: (String) -> NamedDomainObjectProvider<C>) {
```

Comment:
Instead exposing this class to the ABI, the kotlin std provides [PropertyDelegateProvider](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.properties/-property-delegate-provider/), but this interface is not allowed in Gradle public ABI (yet), while it is still a valid Kotlin std class/interface. Any chance to relaxe the ABI validator and allow Kotlin std types in the Gradle Kotlin DSL only?